lib/xdr/ledger_entries/asset_code12.ex

```diff
@@ -45,22 +45,24 @@ defmodule StellarBase.XDR.AssetCode12 do
   @impl true
   def decode_xdr!(bytes, term \\ nil)
 
-  def decode_xdr!(bytes, _term) do
-    {%XDR.FixedOpaque{opaque: code}, rest} =
+  def decode_xdr!(<<bytes::binary-size(12), rest::binary>>, _term) do
+    {%XDR.FixedOpaque{opaque: code}, _rest} =
       XDR.FixedOpaque.decode_xdr!(bytes, opaque_spec(bytes))
 
     {new(code), rest}
   end
 
+  @spec build_opaque(code :: binary(), length :: non_neg_integer()) :: XDR.FixedOpaque.t()
+  defp build_opaque(code, length) do
+    zeros = 12 - byte_size(code)
+    bin = <<code::binary, 0::zeros*8>>
+    XDR.FixedOpaque.new(bin, length + zeros)
+  end
```

Comment:
@CristhianRodriguezMolina IMHO I have these suggestions about this function:
- I think we could replace `12` with the `@max_length` constant.
- Why don't we use `length` instead of calling the `byte_size/1` function?
- Since this function will pad the `code` until 12 bytes, what about replacing `length + zeros` with `@max_length`, couldn't we?

Reply:
Ok, these changes sound good to me.